rulette-mysql-provider/src/main/java/com/github/kislayverma/rulette/mysql/MysqlDataProvider.java

```diff
@@ -102,6 +108,12 @@ public Rule saveRule(String ruleSystemName, Rule rule) throws SQLException, Exce
             ResultSet resultSet = preparedStatement.executeQuery();
 
             List<Rule> ruleList = convertToRules(resultSet, metaData);
+
+
+            resultSet.close();
```

Comment:
@abhinav3: We should do the way it's done in the answer you shared. Try, catch, FINALLY - to ensure we are closing all the resources even if an exception is thrown in between.